### PR TITLE
CS Apollyon Rework

### PR DESCRIPTION
**Pull request by jmcmorris**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Migrate and rework CS Apollyon Limbus area to be in the new battlefield framework. All of this work is heavily based off of the following captures and sources:
https://ffxiclopedia.fandom.com/wiki/CS_Apollyon
https://www.bg-wiki.com/ffxi/CS_Apollyon
https://www.youtube.com/watch?v=NHDH5Nz7MAE
https://www.youtube.com/watch?v=0E4At8VB4nQ
https://www.youtube.com/watch?v=iJgt77S_Af4
https://www.youtube.com/watch?v=qnEGCFWnOns
https://www.youtube.com/watch?v=80n5J0yh7ng

There are some supporting changes that I committed separately so they can easily be viewed distinctly from the core CS Apollyon work.

## Steps to test these changes

```
!addkeyitem red_card
!addkeyitem cosmo_cleanse
!additem 2127
!pos -600 -0.5 -600 38
```

Play through the area following the above guides and captures.

### CODE REVIEW

**Review by claywar — APPROVED** [review]

File: scripts/battlefields/Apollyon/cs_apollyon.lua [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+        local group = boss:getLocalVar("supportGroup")
+        local offsets = group == 0 and supportOffsets or otherSupportOffsets
+        for _, offset in ipairs(offsets) do
+            if GetMobByID(bossID + offset):getStatus() ~= xi.status.DISAPPEAR then
+                return
+            end
+        end
+
+        boss:useMobAbility(603)
```
Comment (WinterSolstice8) [inline scripts/battlefields/Apollyon/cs_apollyon.lua]: Surely, they don't actually use lateral slash which is a tonberry TP move

Consider using `boss:injectActionPacket` to send the action packet to give the client what it needs and a timer to replace the weaponskill listener

If injectActionPacket doesn't do what we need then we can add functionality to do it

Comment (jmcmorris) [reply scripts/battlefields/Apollyon/cs_apollyon.lua]: I didn't know about `injectActionPacket` I'll give that a try. Here's what the `0x028 Action By NPC` packet looks like.
image

File: sql/mob_skills.sql [hunk 2]
```diff
@@ -436,7 +436,7 @@ INSERT INTO `mob_skills` VALUES (599,885,'arctic_impact',1,15.0,2000,2000,4,0,0,
 INSERT INTO `mob_skills` VALUES (600,886,'cold_wave',1,15,2000,2000,4,0,0,0,0,0,0);
 INSERT INTO `mob_skills` VALUES (601,887,'hiemal_storm',1,15.0,2000,2000,4,0,0,0,0,0,0);
 -- INSERT INTO `mob_skills` VALUES (602,346,'hypothermal_combustion',0,7.0,2000,1500,4,0,0,0,0,0,0);
--- INSERT INTO `mob_skills` VALUES (603,432,'lateral_slash',0,7.0,2000,1500,4,0,0,0,0,0,0);
+INSERT INTO `mob_skills` VALUES (603,432,'lateral_slash_nm',0,7.0,2000,0,1,0,0,0,0,0,0);
```
Comment (TeoTwawki) [inline sql/mob_skills.sql]: Strongly suggest a more decretive term appended than `_nm`

Comment (jmcmorris) [reply sql/mob_skills.sql]: Removed this entirely as I'm now using `injectActionPacket`

**Comment by jmcmorris** [comment]
@TeoTwawki How do you feel about this new iteration? The mob scripts call into the battlefield function defined in `cs_apollyon.lua`.

**Comment by TeoTwawki** [comment]
> @TeoTwawki How do you feel about this new iteration? The mob scripts call into the battlefield function defined in `cs_apollyon.lua`.

I think that looks good :+1:

**Review by TeoTwawki — APPROVED** [review]

**Review by WinterSolstice8 — APPROVED** [review]

**Pull request merged**